src/main/java/calculator/StringParser.java

```diff
@@ -0,0 +1,49 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class StringParser {
+	private static final String DEFAULT_DELIMITER = ",:";
+	private static final Pattern PATTERN = Pattern.compile("(//(.+)\\n)?(.*)");
+
+	public List<Integer> parse(String numberSequence) {
+		Matcher matcher = PATTERN.matcher(numberSequence);
+		String delimiters = generateDelimiter(matcher);
+		String numbers = matcher.group(3);
+
+		return Arrays.stream(numbers.split(delimiters))
+			.map(this::parseElement)
+			.collect(Collectors.toList());
+	}
+
+	private String generateDelimiter(Matcher matcher) {
+		StringBuilder sb = new StringBuilder(DEFAULT_DELIMITER);
+
+		if (matcher.find() && Objects.nonNull(matcher.group(2))) {
+			sb.append(matcher.group(2));
+		}
+
+		return new StringBuilder("[]")
+			.insert(1, sb)
+			.toString();
```

Comment:
문자열 연산을 많이하는 것이 아닌데 굳이 StringBuilder 객체를 생성할 필요가 있을까 싶어요. 🤔
지금 같이 간단한 경우에는 아래와 같이 문자열을 `+` 조합해서 사용하는 것으로도 충분해보이네요. 😃

```suggestion
    private String generateDelimiter(Matcher matcher) {
        if (matcher.find() && Objects.nonNull(matcher.group(2))) {
            return "[" + DEFAULT_DELIMITER + matcher.group(2) + "]";
        }

        return "[" + DEFAULT_DELIMITER + "]";
    }
```